apis/tempo/v1alpha1/tempostack_types.go

```diff
@@ -417,6 +417,33 @@ type SearchSpec struct {
 	MaxResultLimit int `json:"maxResultLimit,omitempty"`
 }
 
+// ObjectStorageSecretType defines the type of storage which can be used with the Tempo cluster.
+//
+// +kubebuilder:validation:Enum=s3
+type ObjectStorageSecretType string
+
+const (
+	// ObjectStorageSecretS3 when using S3 for Tempo storage.
+	ObjectStorageSecretS3 ObjectStorageSecretType = "s3"
+)
+
+// ObjectStorageSecretSpec is a secret reference containing name only, no namespace.
+type ObjectStorageSecretSpec struct {
+	// Type of object storage that should be used
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:select:s3"},displayName="Object Storage Secret Type"
+	Type ObjectStorageSecretType `json:"type"`
```

Comment:
What is the default? 

What happens if the type is not specified? Does the CR create fail or is it defaulted to `s3`?